docs/sources/experiments/algorithm.md

```diff
@@ -0,0 +1,182 @@
+- [Feature, label, index, state, user metadata](https://github.com/dssg/triage/blob/master/src/triage/component/architect/planner.py#L89-L112)
+
+### Recap
+
+At this point, all finished matrices and metadata will be saved under the `project_path` supplied by the user to the Experiment constructor, in the subdirectory `matrices`.
+
+## 4. Running Models
+
+The last phase of an Experiment run uses the completed design matrices to train, test, and evaluate classifiers. This procedure writes a lot of metadata to the 'results' schema.
+
+
+### Train
+
+Each matrix marked for training is sent through the configured grid in the experiment's `grid_config`. This works much like the scikit-learn `ParameterGrid` (and in fact uses it on the backend). It cycles through all of the classifiers and hyperparameter combinations contained herein, and calls `.fit()` with that train matrix. Any classifier that adheres to the scikit-learn `.fit/.transform` interface and is available in the Python environment will work here, whether it is a standard scikit-learn classifier, a third-party library like XGBoost, or a custom-built one in the calling repository (for instance, one that implements the problem domain's baseline heuristic algorithm for comparison).  Metadata about the trained classifier is written to the `results.models` Postgres table. The trained model is saved to a filename with the model hash (see Model Hash section below).
+
+#### Model Groups
+
+Each model is assigned a 'model group'. A model group represents a number of trained classifiers that we want to treat as equivalent by some criteria. By default, this is aimed at defining models which are equivalent across time splits, to make analyzing model stability easier. The experiment defines model groups by a static set of data about the model (classifier module, hyperparameters, feature list) and a user-supplied list of keys that must correspond to some key in the matrix metadata (See end of 'Retrieving Data and Saving Completed Matrix' section). This data is stored in the `results.model_groups` table, along with a `model_group_id` that is used as a foreign key in the `results.models` table.
+
+
+#### Model Hash
+Each trained model is assigned a hash, for the purpose of uniquely defining and caching the model. This hash is based on the training matrix metadata, classifier path, hyperparameters (except those which concern execution and do not affect results of the classifier, such as `n_jobs`), and the given project path for the Experiment. This hash can be found in each row of the `results.models` table. It is enforced as a unique key in the table.
+
+#### Global Feature Importance
+The training phase also writes global feature importances to the database, in the `results.feature_importances` table. A few methods are queried to attempt to compute feature importances:
+* The bulk of these are computed using the trained model's `._feature_importances_` attribute, if it exists.
```

Comment:
Is it really `model._feature_importances_`, (with the surrounding underscores)?

Reply:
Oops, just the underscore on the end is real. The first one is a typo.